pkg/api/validation/validation.go

```diff
@@ -441,9 +441,27 @@ func validateHostPathVolumeSource(hostPath *api.HostPathVolumeSource) errs.Valid
 
 func validateGitRepoVolumeSource(gitRepo *api.GitRepoVolumeSource) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
+	target := gitRepo.Directory
+
 	if gitRepo.Repository == "" {
 		allErrs = append(allErrs, errs.NewFieldRequired("repository"))
 	}
+	if path.IsAbs(target) {
```

Comment:
@pmorie Really appreciate your suggestion. Can I do the clean up in another PR? just want to keep this one simple & clean.